src/rest/routes/deferred/deferred-request-parse.ts

```diff
@@ -0,0 +1,19 @@
+import { Request, Response } from "express";
+import { errorWrapper } from "../../helper";
+import {
+	extractSubscriptionDeferredInstallPayload,
+	SubscriptionDeferredInstallPayload
+} from "services/subscription-deferred-install-service";
+import { InvalidArgumentError } from "config/errors";
+
+export const DeferredRequestParseRoute = errorWrapper("ParseRequestId", async function DeferredRequestParseRoute(req: Request, res: Response<SubscriptionDeferredInstallPayload>) {
+	const requestId = req.params.requestId;
+	if (!requestId) {
+		req.log.warn("Missing requestId in query");
+		throw new InvalidArgumentError("Missing requestId in query");
+	}
+
+	const deferredInstallPayload = await extractSubscriptionDeferredInstallPayload(requestId);
```

Comment:
Please cross check that jiraHost from extracted match the one from JwtHandler. Otherwise it could be a security leak that anyone with a requestId (supposed to be public over the network in url) can request information from other jira host.

Reply:
Check added

Reply:
This check will not be required as this path does not go through JwtHandler middleware. This change is done in the FE PR.